### PR TITLE
Vue: Support Vue DOM template formatting

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Alternative #8281, after #8153 

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
This html called [`vue DOM template`](https://vuejs.org/v2/guide/components.html#DOM-Template-Parsing-Caveats)?

File: src/language-html/parser-html.js [hunk 1]
```diff
@@ -105,6 +107,21 @@ function ngHtmlParser(
...
+    recognizeSelfClosing = true;
+    normalizeTagName = true;
+    normalizeAttributeName = true;
+    allowHtmComponentClosingTags = true;
+    isTagNameCaseSensitive = false;
+    const htmlParseResult = parser.parse(input, {
+      canSelfClose: recognizeSelfClosing,
+      allowHtmComponentClosingTags,
+      isTagNameCaseSensitive,
+    });
```
Comment (fisker) [inline src/language-html/parser-html.js:121]: HTML parser `normalizeAttributeName: true`, should we apply?

**Comment by sosukesuzuki** [comment]
> This html called vue DOM template?

I didn't know that but seems to be right. I'll fix changelog.

File: src/language-html/parser-html.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-html/parser-html.js:121]: `normalizeAttributeName` is not parser option. Is is used for AST transformation after parse.

Comment (fisker) [reply src/language-html/parser-html.js:121]: Okay, I don't know that.

**Comment by fisker** [comment]
Let's use this as changelog example? Stable don't throw, but we change it to throw error kind of weird.


**Input:**
```vue
<!DOCTYPE html>
<html>
  <body STYLE="color: #333">
    <DIV id="app">
      <DIV>First Line</DIV><DIV>Second Line</DIV>
    </DIV>
    <script src="https://unpkg.com/vue/dist/vue.js"></script>
    <script>
      new Vue({
        el: "#app",
        data() {
          return {}
        },
      });
    </script>
  </body>
</html>
```

**Stable:**
```vue
<!DOCTYPE html>
<html>
  <body STYLE="color: #333">
    <DIV id="app"> <DIV>First Line</DIV><DIV>Second Line</DIV> </DIV>
    <script src="https://unpkg.com/vue/dist/vue.js"></script>
    <script>
      new Vue({
        el: "#app",
        data() {
          return {};
        },
      });
    </script>
  </body>
</html>

```

**Master:**
```vue
<!DOCTYPE html>
<html>
  <body style="color: #333">
    <div id="app">
      <div>First Line</div>
      <div>Second Line</div>
    </div>
    <script src="https://unpkg.com/vue/dist/vue.js"></script>
    <script>
      new Vue({
        el: "#app",
        data() {
          return {};
        },
      });
    </script>
  </body>
</html>

```

**Comment by sosukesuzuki** [comment]
> Let's use this as changelog example? Stable don't throw, but we change it to throw error kind of weird.

Looks good. I fixed changelog.

**Pull request merged**